src/Planners/src/UnicyclePlanner.cpp

```diff
@@ -0,0 +1,520 @@
+/**
+ * @file UnicyclePlanner.h
+ * @authors Diego Ferigo, Stefano Dafarra
+ * @copyright 2021 Istituto Italiano di Tecnologia (IIT). This software may be modified and
+ * distributed under the terms of the GNU Lesser General Public License v2.1 or any later version.
+ */
+
+#include "BipedalLocomotion/Planners/UnicyclePlanner.h"
+#include "BipedalLocomotion/TextLogging/Logger.h"
+
+#include <FootPrint.h>
+#include <UnicyclePlanner.h>
+#include <iDynTree/Core/VectorFixSize.h>
+
+#include <cassert>
+#include <optional>
+
+using namespace BipedalLocomotion;
+
+class Planners::UnicyclePlanner::Impl
+{
+public:
+    struct
+    {
+        double planner;
+        double integrator;
+    } dt;
+
+    struct
+    {
+        double t0;
+        double tf;
+    } horizon;
+
+    struct
+    {
+        struct
+        {
+            double x = 0.10;
+            double y = 0.00;
+        } reference;
+
+        struct
+        {
+            double unicycle = 10.0;
+            double slowWhenTurning = 0.0;
+        } gains;
+    } controller;
+
+    struct
+    {
+        double time = 1.0;
+        double position = 1.0;
+    } weights;
+
+    struct
+    {
+        double min;
+        double max;
+        double nominal;
+    } duration;
+
+    struct
+    {
+        double min;
+        double max;
+    } stepLength;
+
+    struct
+    {
+        double min;
+        double nominal;
+    } feetDistance;
+
+    struct
+    {
+        double min;
+        double max;
+    } angleVariation;
+
+    struct
+    {
+        double stancePhaseRatio;
+        bool startWithLeft = false;
+        bool terminalStep = true;
+        bool resetStartingFootIfStill = false;
+    } gait;
+
+    UnicyclePlannerOutput outputRef;
+    std::optional<UnicyclePlannerOutput> output = std::nullopt;
+
+    std::shared_ptr<::FootPrint> left;
+    std::shared_ptr<::FootPrint> right;
+    std::unique_ptr<::UnicyclePlanner> planner;
+};
+
+Planners::UnicyclePlanner::UnicyclePlanner()
+    : m_pImpl{std::make_unique<Impl>()}
+{
+}
+
+Planners::UnicyclePlanner::~UnicyclePlanner() = default;
+
+bool Planners::UnicyclePlanner::initialize(
+    std::weak_ptr<const ParametersHandler::IParametersHandler> handler)
+{
+    constexpr auto logPrefix = "[UnicyclePlanner::initialize]";
+
+    auto ptr = handler.lock();
+
+    if (ptr == nullptr)
+    {
+        log()->error("{} The handler has to point to an already initialized IParametersHandler.",
+                     logPrefix);
+        return false;
+    }
+
+    bool okPlanner = true;
+    m_pImpl->planner = std::make_unique<::UnicyclePlanner>();
+
+    // ==
+    // dt
+    // ==
+
+    if (!ptr->getParameter("sampling_time", m_pImpl->dt.planner))
```

Comment:
Is this value used? If so, why it is different from ``sampling_time_integrator``?

Reply:
Should it be the same all the times? There are two different methods that take the dt: `UnicyclePlanner::setPlannerPeriod` and `UnicyclePlanner::setMaximumIntegratorStepSize`. In the previous usage, the same dt is used. However, I thought to expose the two different parameters.

Reply:
But, if I understood well, only one is used at the moment, right?

Reply:
Both are set here:

https://github.com/diegoferigo/bipedal-locomotion-framework/blob/ebb2d118ef3a37948d680d3eda7945d7b931e789/src/Planners/src/UnicyclePlanner.cpp#L141-L142

Reply:
I still don't see where ``m_pImpl->dt.planner`` is used :grimacing: But in any case, it is better to force them equal.

Reply:
Do you know when you're sure you did something but your brain tricks you not to see things in front of your face. Of course I had in mind:

```cpp
    okPlanner = okPlanner && m_pImpl->planner->setPlannerPeriod(m_pImpl->dt.planner);
    okPlanner = okPlanner && m_pImpl->planner->setMaximumIntegratorStepSize(m_pImpl->dt.integrator);
```

I'll then remove the integrator time step and just use the planner period.

Reply:
Nice!

Reply:
Done in https://github.com/dic-iit/bipedal-locomotion-framework/compare/d6e7ed3657366daf638174c9bfedba2e86f306c4..7dfc4ced78be78f7981004dab9e0d7d7779ebae2